packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5880,17 +5880,18 @@ class _PointerListener extends SingleChildRenderObjectWidget {
 class MouseRegion extends StatefulWidget {
   /// Creates a widget that forwards mouse events to callbacks.
   ///
-  /// By default, all callbacks are empty, `cursor` is unset, and `opaque` is
-  /// `true`.
+  /// By default, all callbacks are empty, [cursor] is deferred, and [opaque] is
+  /// true.
   const MouseRegion({
     Key key,
     this.onEnter,
     this.onExit,
     this.onHover,
-    this.cursor,
+    MouseCursor cursor,
```

Comment:
Why do we allow null here? Shouldn't this just be

```suggestion
    this.cursor = DeferredMouseCursor.instance,
```

with an assert that cursor is not null?

Reply:
Where do we draw the line of widgets that don't allow null values and those that allow null values and calculate default values in the implementation? My plan here is that all widgets and render objects allow null belong to the 2nd category so that the design looks more consistent. Suggestion is welcome.

Reply:
In my opinion all the basic widgets in `widgets/` should use regular default values to indicate that this value is very much not computed. Only the them-able widgets in Material/Cupertino need the other behavior.

Reply:
I agree. Widgets in the widgets library typically have  conventional defaults.